pkgs/development/haskell-modules/configuration-common.nix

```
@@ -1330,6 +1330,10 @@ self: super: {
   # https://github.com/ennocramer/monad-dijkstra/issues/4
   monad-dijkstra = dontCheck (doJailbreak super.monad-dijkstra);
 
+  libsodium = overrideCabal super.libsodium (drv: {
+    libraryToolDepends = drv.libraryToolDepends or [self.c2hs];
```

I think something like this would be slightly safer:

```suggestion
    libraryToolDepends = (drv.libraryToolDepends or []) ++ [self.c2hs];
```